[V3] Adding wui-text component

svenvoskamp opened the pull request:
# Changes

- feat: Add `wui-text` component 

# Description

We use the following text styles in V3:

Large
Medium
Small
XXSmall
Medium Numerals

Took them from Figma and implemented in the component. 


Link to Storybook: ...

0xAsimetriq commented on packages/ui/src/utils/ThemeUtil.ts:
```diff
@@ -13,6 +13,32 @@ export const globalStyles = css`
+    --wui-line-height-md: 125%;
+    --wui-line-height-lg: 130%;
+
+    --wui-color-accent: #47a1ff;
```
Color names don't seem to match ones in figma's `Library 3.0` file

0xAsimetriq commented on packages/ui/src/components/wui-text/styles.ts:
```diff
@@ -0,0 +1,94 @@
+
+  .wui-color-accent {
+    color: var(--wui-color-accent);
+  }
```
Inconsistent spacing after `}`

0xAsimetriq commented on packages/ui/src/components/wui-text/index.ts:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    return html`
+      <span>
```
Not sure about this, but is it possible to remove span and style `slot` element directly?

0xAsimetriq commented on packages/ui/src/components/wui-text/index.ts:
```diff
@@ -0,0 +1,63 @@
+
+@customElement('wui-text')
+export class WuiText extends LitElement {
+  public static styles = [globalStyles, styles]
```
Something to research in `lit`, do you think we can find a better solution to global styles i.e. instead of defining them in each component, maybe we can define one time somewhere?

0xAsimetriq reviewed (changes requested):
Left few comments

svenvoskamp commented:
@0xAsimetriq 

- Renamed colors
- Inconsistent spaces fixed
- Removed span in `wui-text`
- Did some research on global `css` styles for `lit` but couldn't find a better solution than what we currently have.

0xAsimetriq reviewed (approved):
Looks good to me 👍

0xAsimetriq commented on apps/gallery/.storybook/preview.tsx:
```diff
@@ -0,0 +1,32 @@
+import { Controls, Description, Primary, Source, Stories, Subtitle, Title } from '@storybook/blocks'
+import React from 'react'
+
+import {} from '@storybook/blocks'
```
Is this import needed?

0xAsimetriq commented on apps/gallery/.storybook/preview.tsx:
```diff
@@ -0,0 +1,32 @@
+import { Controls, Description, Primary, Source, Stories, Subtitle, Title } from '@storybook/blocks'
+import React from 'react'
```
Is react import needed? (might be global like in nextjs)

svenvoskamp replied:
Yes it is needed, otherwise you get an `ReferenceError: React is not defined` error

0xAsimetriq commented on packages/ui/src/components/wui-text/index.ts:
```diff
@@ -0,0 +1,71 @@
+    }
+
+    return html`
+      <span>
```
Didn't we remove span? styles file seems to target host now, so is this span element needed?

0xAsimetriq commented on packages/ui/src/utils/ThemeUtil.ts:
```diff
@@ -13,6 +14,33 @@ export const globalStyles = css`
     -webkit-tap-highlight-color: transparent;
     backface-visibility: hidden;
+
+    --wui-font-family: 'SF Pro Text', 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', sans-serif;
```
I think variables should be moved to `:root {}`
